source/js/foundation/pages/youtube-regrets/regrets-reporter-timeline.js

```diff
@@ -0,0 +1,91 @@
+import utility from "../../../utility.js";
+
+let elements = {
+  timeline: `#view-youtube-regrets-reporter .timeline-wrapper .timeline`,
+  segments: `#view-youtube-regrets-reporter .timeline-wrapper .timeline .segment`,
+  button: `#view-youtube-regrets-reporter .timeline-wrapper .btn-timeline-next`,
+};
+
+class RegretsReporterTimeline {
+  constructor() {
+    this.buttonClickCounter = 0;
+
+    this.init();
+  }
+
+  /**
+   * Check if segment is fully shown in current viewport
+   */
+  checkIfInView(seg) {
+    let windowWidth = window.innerWidth || document.documentElement.clientWidth;
+
+    return seg.getBoundingClientRect().right <= windowWidth;
+  }
+
+  /**
+   * Update each timeline segment's opacity
+   */
+  updateOpacity() {
+    elements.segments.forEach((seg) => {
+      if (this.checkIfInView(seg)) {
+        seg.classList.remove(`faded`);
+      } else {
+        seg.classList.add(`faded`);
+      }
+    });
+  }
+
+  /**
+   * Initiate RegretsReporter timeline
+   */
+  init() {
+    this.loadAttempt = this.loadAttempt || 1;
+
+    if (!utility.checkAndBindDomNodes(elements, true)) {
+      if (this.loadAttempt++ < 5) {
+        setTimeout(() => this.init(), 200);
+      }
+      return;
+    }
+
+    this.timeline = elements.timeline[0];
+
+    this.updateOpacity();
+
+    this.timeline.addEventListener("transitionend", () => {
+      this.updateOpacity();
+    });
+
+    window.addEventListener("resize", () => {
+      this.updateOpacity();
+    });
+
+    elements.button[0].addEventListener("click", () => {
+      let segmentWidth = elements.segments[0].offsetWidth;
+      let windowWidth =
+        window.innerWidth || document.documentElement.clientWidth;
+      let lastItemShown = this.checkIfInView(
+        document.querySelector(".timeline .segment:last-of-type")
```

Comment:
probably worth caching this element in a `const` outside the event listener? Also, we already have `timeline`, so we can query that directly:
```
const last = timeline.querySelector(`.segment:last-of-type`);
```

Reply:
👍